lib/src/app/navigation/route_page_manager.dart

```diff
@@ -0,0 +1,148 @@
+import 'package:allthenews/src/app/navigation/app_path.dart';
+import 'package:allthenews/src/app/navigation/app_router_information_parser.dart';
+import 'package:allthenews/src/ui/pages/authentication/login/login_page.dart';
+import 'package:allthenews/src/ui/pages/authentication/registration/registration_page.dart';
+import 'package:allthenews/src/ui/pages/dashboard/dashboard_page.dart';
+import 'package:allthenews/src/ui/pages/dashboard/news/latest/latest_news_page.dart';
+import 'package:allthenews/src/ui/pages/dashboard/news/most_popular/most_popular_news_page.dart';
+import 'package:allthenews/src/ui/pages/home/bottom_bar_notifier.dart';
+import 'package:allthenews/src/ui/pages/home/home_page.dart';
+import 'package:allthenews/src/ui/pages/location/location_page.dart';
+import 'package:allthenews/src/ui/pages/presentation/presentation_page.dart';
+import 'package:allthenews/src/ui/pages/profile/profile_page.dart';
+import 'package:allthenews/src/ui/pages/settings/settings_page.dart';
+import 'package:allthenews/src/ui/pages/web_view/web_view_page.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+class RoutePageManager extends ChangeNotifier {
+  final BottomBarNotifier _bottomBarNotifier = BottomBarNotifier();
+  final List<Page> _pages = [];
+
+  RoutePageManager() {
+    _bottomBarNotifier.addListener(notifyListeners);
+    _pages.addAll(_buildInitialPages(_bottomBarNotifier));
+  }
+
+  List<Page> _buildInitialPages(BottomBarNotifier state) {
+    return [
+      MaterialPage(
+        child: HomePage(state),
+        key: const ValueKey(PagePathLocation.homePage),
+      ),
+      MaterialPage(
+        child: PresentationPage(),
+        key: const ValueKey(PagePathLocation.presentationPage),
+      )
+    ];
+  }
```

Comment:
czy to jest potrzebne? to nie jest tak, że presentation page doda ścieżkę do `HomePage` ?

Reply:
HomePage ustawiłem jako początek stosu i Presentation tak naprawdę usuwa się ze stosu i zostawia na wierzchu HomePage. Ustawiłem tak, żeby zawsze kiedy HomePage jest na wierzchu to, żeby nie było możliwości, że pod spodem jest coś innego, jako że HomePage stanowi bazowy widok